pep_sphinx_extensions/pep_zero_generator/pep_0_parser.py

```diff
@@ -0,0 +1,288 @@
+
+        # Title
+        self.title: str = metadata["Title"]
+
+        # Type
+        self.pep_type: str = metadata["Type"]
+        if self.pep_type not in self.type_values:
+            self.raise_pep_error(f"{self.pep_type} is not a valid Type value", pep_num=True)
+
+        # Status
+        status = metadata["Status"]
+        if status not in self.status_values:
+            if status == "April Fool!":  # See PEP 401 :)
+                status = "Rejected"
+            else:
+                self.raise_pep_error(f"{status} is not a valid Status value", pep_num=True)
+
+        # Special case for Active PEPs.
+        if status == "Active" and self.pep_type not in {"Process", "Informational"}:
+            msg = "Only Process and Informational PEPs may have an Active status"
+            self.raise_pep_error(msg, pep_num=True)
+
+        # Special case for Provisional PEPs.
+        if status == "Provisional" and self.pep_type != "Standards Track":
+            msg = "Only Standards Track PEPs may have a Provisional status"
+            self.raise_pep_error(msg, pep_num=True)
+        self.status: str = status
+
+        # Parse PEP authors
+        self.authors: list[Author] = self.parse_authors(metadata["Author"], author_lookup)
+
+    def parse_authors(self, author_header: str, author_lookup: dict) -> list[Author]:
+        """Parse Author header line"""
+        authors_and_emails = self._parse_author(author_header)
+        if not authors_and_emails:
+            raise self.raise_pep_error("no authors found", pep_num=True)
+        return [Author(author_tuple, author_lookup) for author_tuple in authors_and_emails]
+
+    angled = re.compile(r"(?P<author>.+?) <(?P<email>.+?)>(,\s*)?")
+    paren = re.compile(r"(?P<email>.+?) \((?P<author>.+?)\)(,\s*)?")
+    simple = re.compile(r"(?P<author>[^,]+)(,\s*)?")
+
+    @staticmethod
+    def _parse_author(data: str) -> list[tuple[str, str]]:
+        """Return a list of author names and emails."""
+        # XXX Consider using email.utils.parseaddr (doesn't work with names
+        # lacking an email address.
+
+        author_list = []
+        for regex in (PEP.angled, PEP.paren, PEP.simple):
+            for match in regex.finditer(data):
+                # Watch out for suffixes like 'Jr.' when they are comma-separated
+                # from the name and thus cause issues when *all* names are only
+                # separated by commas.
+                match_dict = match.groupdict()
+                author = match_dict["author"]
+                if not author.partition(" ")[1] and author.endswith("."):
+                    prev_author = author_list.pop()
+                    author = ", ".join([prev_author, author])
+                if "email" not in match_dict:
+                    email = ""
+                else:
+                    email = match_dict["email"]
+                author_list.append((author, email))
+            else:
```

Comment:
This else is going to be hitted always because there is no break in the loop

Reply:
I think it's intended to break out of the parent loop, but agree written oddly. Will simplify a little

Reply:
done